app/models/label/LabelTable.scala

```diff
@@ -1012,6 +1014,22 @@ object LabelTable {
     _labels.list.map(l => LabelLocationWithSeverity(l._1, l._2, l._3, l._4.get, l._5.get, l._6, l._7, l._8, l._9, l._10))
   }
 
+  /**
+   * Returns all the submitted labels as well as the tags that were used with each label and counts of each tag.
+   */
+  def selectTagsOfLabels(regionIds: List[Int]): List[LabelWithTags] = db.withSession { implicit session =>
+    val _labels = for {
+      _l <- labels
+      _lType <- labelTypes if _l.labelTypeId === _lType.labelTypeId
+      _lPoint <- labelPoints if _l.labelId === _lPoint.labelId
+    } yield (_lType.labelType, _l.tags.unnest)
+
+    val _labelsList = _labels.list
```

Comment:
Ooo just realized that you're calling `_labels.list` here. When you call `.list`, that runs the query above on the database, thus loading all of the labels into memory :( Hopefully you can run all the `groupBy` stuff _before_ calling `.list`! Most of the syntax is the same.